pages/exercises/[lessonSlug].tsx

```diff
@@ -45,14 +80,113 @@ const Exercises: React.FC<QueryDataProps<GetAppQuery>> = ({ queryData }) => {
   ]
 
+  const exercise = mockExercises[exerciseIndex]
+
   return (
     <Layout title={currentLesson.title}>
+      {exercise ? (
+        <Exercise
+          exercise={exercise}
+          setExerciseIndex={setExerciseIndex}
+          lessonTitle={currentLesson.title}
+          showPreviousButton={exerciseIndex > 0}
+          showSkipButton={exerciseIndex < mockExercises.length - 1}
+        />
+      ) : (
+        <ExerciseList
+          tabs={tabs}
+          setExerciseIndex={setExerciseIndex}
+          lessonTitle={currentLesson.title}
+        />
+      )}
+      {alerts && <AlertsDisplay alerts={alerts} />}
+    </Layout>
+  )
+}
+
+type ExerciseProps = {
+  exercise: ExerciseCardProps
+  setExerciseIndex: React.Dispatch<React.SetStateAction<number>>
+  lessonTitle: string
+  showPreviousButton: boolean
+  showSkipButton: boolean
+}
+
+const Exercise = ({
+  exercise,
+  setExerciseIndex,
+  lessonTitle,
+  showPreviousButton,
+  showSkipButton
+}: ExerciseProps) => {
+  return (
+    <div className="w-75 mx-auto">
+      <button
+        className="btn ps-0 d-flex align-items-center"
+        onClick={() => setExerciseIndex(-1)}
+      >
+        <ArrowLeftIcon size="medium" />
+        <div className="p-3">
+          <Text size="xs">Exit</Text>
+        </div>
+      </button>
+
+      <h1 className="mb-4 fs-2">{lessonTitle}</h1>
+      <ExerciseCard
+        challengeName={exercise.challengeName}
+        problem={exercise.problem}
+        answer={exercise.answer}
+        explanation={exercise.explanation}
+      />
+      <div className="d-flex justify-content-between mt-4">
+        {showPreviousButton ? (
+          <button
+            onClick={() => setExerciseIndex(i => i - 1)}
+            className="btn btn-outline-primary"
+          >
+            PREVIOUS
+          </button>
+        ) : (
+          <div />
+        )}
+        {showSkipButton ? (
+          <button
+            onClick={() => setExerciseIndex(i => i + 1)}
+            className="btn btn-outline-primary"
+          >
+            SKIP
+          </button>
+        ) : (
+          <div />
+        )}
+      </div>
+    </div>
+  )
+}
+
+type ExerciseListProps = {
+  tabs: { text: string; url: string }[]
+  setExerciseIndex: React.Dispatch<React.SetStateAction<number>>
+  lessonTitle: string
+}
+
+const ExerciseList = ({
+  tabs,
+  setExerciseIndex,
+  lessonTitle
+}: ExerciseListProps) => {
+  return (
+    <>
       <NavCard
         tabSelected={tabs.findIndex(tab => tab.text === 'exercises')}
         tabs={tabs}
       />
-      <h1 className="my-4">{currentLesson.title}</h1>
-      {alerts && <AlertsDisplay alerts={alerts} />}
+      <div className="d-flex justify-content-between align-items-center">
+        <h1 className="my-4 fs-2">{lessonTitle}</h1>
+        <NewButton onClick={() => setExerciseIndex(0)}>
+          SOLVE EXERCISES
+        </NewButton>
+      </div>
       <div className="container">
         <div className="row">
```

Comment:
After connecting it with the backend, it'll then be updated to show many exercises (20-30) in a good layout?

Reply:
Yes.